stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -138,15 +139,41 @@ internal class PaymentOptionsActivity : BasePaymentSheetActivity<PaymentOptionRe
                 }
             }
         }
+    }
+
+    private val label: String
+        get() = resources.getString(
+            R.string.stripe_paymentsheet_add_button_label
+        )
 
-        viewModel.userSelection.observe(this) { paymentSelection ->
-            if (paymentSelection != null) {
-                onActionCompleted(paymentSelection)
+    @VisibleForTesting
+    private val viewStateObserver:  (ViewState.Add?, DefaultPrimaryButton) -> Unit
+        get() = { state, addButton ->
+            addButton.updateState(state)
+            if (state != null) {
+                when (state) {
+                    is ViewState.Add.Ready -> {
+                        addButton.setReady()
+                    }
+                    ViewState.Add.Confirming -> {
+                        addButton.setConfirm()
+                    }
+                    is ViewState.Add.Completed -> {
+                        addButton.setCompleted {
+                            onActionCompleted(state.result)
+                        }
+                    }
```

Comment:
I feel like `PrimaryButton` could have its own `State` type that we map `ViewState.Buy` and `ViewState.Add` to. Then we can do something like this instead of explicitly calling `setReady()`, `setConfirm()`, and `setCompleted()`.

```
class PrimaryButton {
  fun updateState(state: PrimaryButton.State) {
    when (state) {
      // handle cases
    }
  }
}
```